src/main/java/org/opensearch/knn/index/engine/Encoder.java

```diff
@@ -36,4 +36,14 @@ default String getName() {
      *          return {@link CompressionLevel#NOT_CONFIGURED}
      */
     CompressionLevel calculateCompressionLevel(MethodComponentContext encoderContext, KNNMethodConfigContext knnMethodConfigContext);
+
+    /**
+     * Validates config of encoder
+     *
+     * @return Validation output of encoder parameters
+     */
+    default TrainingConfigValidationOutput validateEncoderConfig(TrainingConfigValidationInput validationInput) {
```

Comment:
Did we miss something here ? I don't see the validationInput used in the builder ?